web/src/stream_popover.js

```diff
@@ -619,6 +620,56 @@ export async function build_move_topic_to_stream_popover(
         },
         post_render: move_topic_post_render,
     });
+
+    let default_message_move_count;
+    // The "if" part covers the case when "Move topic" is clicked,
+    // while the "else" part accounts for the case when "Move messages" is clicked.
+    if (args.from_message_actions_popover) {
+        default_message_move_count = message_util.get_messages_in_topic_from_here(
+            current_stream_id,
+            topic_name,
+            message.id,
+        ).length;
+    } else {
+        default_message_move_count = message_util.get_messages_in_topic(
+            current_stream_id,
+            topic_name,
+        ).length;
+    }
+
+    $("#move_messages_count").text(
+        $t(
+            {
+                defaultMessage:
+                    "{count, plural, one {# message} other {# messages}} will be moved.",
+            },
+            {count: default_message_move_count},
+        ),
+    );
+
+    $("#message_move_select_options").on("change", function () {
+        const selected_option = $(this).val();
+        let message_move_count;
+        if (selected_option === "change_one") {
+            message_move_count = 1;
+        } else if (selected_option === "change_later") {
+            message_move_count = default_message_move_count;
+        } else {
+            message_move_count = message_util.get_messages_in_topic(
+                current_stream_id,
+                topic_name,
+            ).length;
+        }
+        $("#move_messages_count").text(
+            $t(
+                {
+                    defaultMessage:
+                        "{count, plural, one {# message} other {# messages}} will be moved.",
+                },
+                {count: message_move_count},
```

Comment:
I guess this code should be in the `post_render` function.

Also, would be good to not re-use `default_message_move_count` as more messages can be received while user changes the option. So, instead would be good to just create a function which retruns count based on the selected option which can be used for both the cases - while setting the text initially and changing the text with options.